File: stack/stack.py
class Stack:
    def __init__(self, first_element=None):
        self.head = first_element
        self.tail = None

    def push(self, data):
        if self.head is None:
            self.head = data
            return None
        if self.tail:
            self.tail.push(data)
        else:
            self.tail = Stack(data)

    def pop(self):
        if self.tail:
            if self.tail.tail:
                return self.tail.pop()
            else:
                temp = self.tail.head
                self.tail = None
                return temp
        else:
            temp = self.head
            self.head = None
            return temp


a = Stack()
a.push(1)
a.push(2)
print(a.pop())
print(a.pop())
print(a.head)

File: stack/stack.js
class Stack {
    constructor(first_element=null) {
        this.head = first_element
        this.tail = null
    }

    push(data) {
        if (this.head == null) {
            this.head = data
            return null
        }
        if (this.tail) {
            this.tail.push(data)
        } else {
            this.tail = new Stack(data)
        }
    }

    pop() {
        var temp
        if (this.tail) {
            if (this.tail.tail) {
                return this.tail.pop()
            }
            else {
                temp = this.tail.head
                this.tail = null
                return temp
            }
        }
        else {
            temp = this.head
            this.head = null
            return temp
        }
    }
}

var stack = new Stack()
stack.push(1)
stack.push(2)
console.log(stack.pop())
console.log(stack.pop())
console.log(stack.pop())